components/dashboard/src/hooks/useStripe.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { useLocation } from "react-router";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+
+interface hasId {
+    id: string;
+}
+
+type PendingStripeSubscription = { pendingSince: number };
+
+type returnType = [
+    boolean,
+    string | undefined,
+    boolean,
+    boolean,
+    string | undefined,
+    number | undefined,
+    (newLimit: number) => Promise<void>,
+];
+
+const useStripe = (subject: hasId | undefined, attributionId: AttributionId, localStorageKey: string): returnType => {
```

Comment:
Suggestion: Instead of making `localStorageKey` a parameter, how about generating it automatically based on the `attributionId`?

E.g. instead of `pendingStripeSubscriptionForUser${user.id}` / `pendingStripeSubscriptionForTeam${team.id}`, we could have a unified `pendingStripeSubscriptionFor${attributionId}` (or the rendered version I guess)

Reply:
yes, that's a better approach.

`localStorageKey` is now calculated in `UsageBasedBillingConfig` .